thirdparty/faiss/faiss/IndexIVF.h

```diff
@@ -89,6 +90,30 @@ struct InvertedListScanner;
 struct IndexIVFStats;
 struct CodePacker;
 
+struct IVFIteratorWorkspace {
```

Comment:
Please add very detailed comments on what this structure does exactly and how it is used